users/vieting/jobs/fairseq.py

```diff
@@ -109,38 +103,95 @@ def create_tsv_and_labels(self):
 
                 rel_audio_path = os.path.relpath(audio_path, common_dir)
                 frames = soundfile.info(audio_path).frames
-
-                # determine whether to write to train or valid
-                if rand.random() >= self.valid_percent:
-                    tsv_out = train_tsv
-                    ltr_out = train_ltr
-                    wrd_out = train_wrd
-                else:
-                    tsv_out = valid_tsv
-                    ltr_out = valid_ltr
-                    wrd_out = valid_wrd
                 
                 # write audio path to tsv files
-                print(f"{rel_audio_path}\t{frames}", file=tsv_out)
+                print(f"{rel_audio_path}\t{frames}", file=dest_tsv)
 
                 # write transcription to transcription files
                 print(
                     " ".join(list(audio_trans.replace(" ", "|"))) + " |",
-                    file=ltr_out,
+                    file=dest_ltr,
                 )
-                print(audio_trans, file=wrd_out)
-        
-        # close all files
-        valid_tsv.close()
-        valid_ltr.close()
-        valid_wrd.close()
-        
-        train_tsv.close()
-        train_ltr.close()
-        train_wrd.close()
+                print(audio_trans, file=dest_wrd)
+
+        dest_tsv.close()
+        dest_ltr.close()
+        dest_wrd.close()
 
+    def get_common_dir(self):
+        """
+        Returns the common directory of all audios given in the corpora.
+        """
+        common_dir = None
+        # iterate over all corpora
+        for corpus_path in self.corpus_paths:
+            corpus_object = corpus.Corpus()
+            corpus_object.load(corpus_path.get())
+            for segment in corpus_object.segments():
+                audio_path = segment.recording.audio
+                assert os.path.exists(audio_path), f"Path {audio_path} does not exist."
+                if common_dir is None:
+                    common_dir = os.path.dirname(audio_path)
+                else:
+                    common_dir = os.path.commonpath([common_dir, os.path.dirname(audio_path)])
```

Comment:
This should yield the same result, however requires probably more memory (since we would need to store every `audio_path` from the corpus, which might be quite many). I don't know how neglectable this effect is, especially when possibly using larger corpora later

Reply:
I don't think it's too critical. You have to load each corpus individually anyway and just keep some string for each recording (which is a small fraction of the total memory). I think it's not realistic that the number of corpora would be so large that this would have an effect here.